src/services/pcn-router/src/Ports.cpp

```diff
@@ -181,6 +181,18 @@ void Ports::removeEntry(Router &parent, const std::string &name) {
 
   auto router_port = parent.get_hash_table<uint16_t, r_port>("router_port");
 
+  // remove the ArpEntry from the datapath, for this port
+  auto arp_table = parent.get_hash_table<uint32_t, arp_entry>("arp_table");
+  auto arp_entries = arp_table.get_all();
+  for (auto &entry : arp_entries) {
+    auto key = entry.first;
+    auto value = entry.second;
+    auto port = parent.get_port(value.port);
+
+    if (port->name() == name)
+      arp_table.remove(key);
+  }
+
```

Comment:
It is not necessary to call `get_port` for each entry, we can compare if the entry is related to this port by using the port's index. Notice that `port` refers to the port that is going to be deleted.

```
if (port->index() == value.port) {
  arp_table.remove(key);
}
```

Reply:
Ok, thanks @mauriciovasquezbernal.